pkg/broker/handler/pool/pool.go

```diff
@@ -27,25 +30,86 @@ type SyncPool interface {
 	SyncOnce(ctx context.Context) error
 }
 
+type healthChecker struct {
+	mux              sync.RWMutex
+	lastReportTime   time.Time
+	maxStaleDuration time.Duration
+}
+
+func (c *healthChecker) reportHealth() {
+	c.mux.Lock()
+	defer c.mux.Unlock()
+	c.lastReportTime = time.Now()
+}
+
+func (c *healthChecker) lastTime() time.Time {
+	c.mux.RLock()
+	defer c.mux.RUnlock()
+	return c.lastReportTime
+}
+
+func (c *healthChecker) start(ctx context.Context) {
+	c.reportHealth()
+	// TODO: allow changing port? It's pure internal though.
```

Comment:
Using a fixed port can lead to flaky tests if the port is not available during tests.

Reply:
You mean in unit test? Unless we assign random port in the test you're referring, it won't make a difference right?

Reply:
Yeah in unit tests. And it's pretty easy to run into that.

You can make the http server a dependency and use httptest.NewServer in tests while still using a fixed port in real code.

Reply:
I still think it makes sense to have the sync pool own the lifecycle of the health check server. Using httptest.Server makes it difficult to do that (it's not an interface). So instead I allow it to change health check port.